### PR TITLE
Integrate with Preact-only runtime

### PR DESCRIPTION
**Pull request by KevinSjoberg**
This is a first attempt to make Mint compatible with the `preact` Mint runtime (meaning no `preact/compat`).

See https://github.com/mint-lang/mint-runtime/pull/8 for details.

### CODE REVIEW

File: core/source/Test/Html.mint [hunk 1]
```diff
@@ -10,9 +10,8 @@ module Test.Html {
     (() => {
       let root = document.createElement('div')
       document.body.appendChild(root)
-      ReactDOM.render(#{node}, root)
+      _render(#{node}, root)
       return new TestContext(root, () => {
-        ReactDOM.unmountComponentAtNode(root)
```
Comment (KevinSjoberg) [inline core/source/Test/Html.mint]: Not too sure about this. I mentioned something similar in https://github.com/mint-lang/mint-runtime/pull/8#issuecomment-588424638.

**Comment by gdotdesign** [comment]
@KevinSjoberg is there something to do here? The unmount thing is the only one thing that still needs to be resolved?

**Comment by KevinSjoberg** [comment]
@gdotdesign still waiting for feedback on the runtime pull request and so I’ve waited with any changes necessary here. I’ll make sure to update this as well whenever the runtime pull request is ready. 🙂

File: core/source/Test/Html.mint (hunk 1 above)
Comment (gdotdesign) [reply core/source/Test/Html.mint]: We need to keep this functionality somehow so tests are run in a clean state: components needs to unsubscribe from providers and stores (which runs in `componentWillUnmount`) does Preact has something which can do this?

Comment (gdotdesign) [reply core/source/Test/Html.mint]: I think I found out how they do it: https://github.com/preactjs/preact/blob/master/compat/src/index.js#L67

**Comment by gdotdesign** [comment]
I've kept the original interface so only the runtime changed :arrow_right:  https://github.com/mint-lang/mint/commit/dc31b6a412377b09d627bd17b3a66c3cb58cefa2

**Pull request closed without merging**